src/main/public/js/postcode-lookup.js

```diff
@@ -0,0 +1,134 @@
+          var addressDetails = this.value.split(', ')
+          var addressSelector = postcodeLookupWidget.querySelector('.address')
+          addressSelector.querySelector('.address-line1').value = addressDetails[0]
+          addressSelector.querySelector('.address-line2').value = addressDetails[1]
+          addressSelector.querySelector('.address-town-or-city').value = addressDetails[2]
+          addressSelector.querySelector('.postcode').value = addressDetails[3]
+          showAddressEntry(postcodeLookupWidget)
+        })
+
+      if (isAnyAddressFieldPopulated(postcodeLookupWidget)) {
+        showAddressEntry(postcodeLookupWidget)
+      }
+    })
+})
+
+function isNorthernIrelandPostcode (postcode) {
+  return postcode.startsWith('BT')
+}
+
+function showAddressEntry (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.address').classList.remove('js-hidden')
+}
+
+function clearPostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('select').innerHTML = ''
+}
+
+function hidePostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.postcode-address-picker').classList.add('hidden')
+}
+
+function showPostcodeDropdown (postcodeLookupWidget) {
+  postcodeLookupWidget.querySelector('.postcode-address-picker').classList.remove('hidden')
+}
+
+function clearAddressFields (postcodeLookupWidget) {
+  var addressSelector = postcodeLookupWidget.querySelector('.address')
+  addressSelector.querySelector('.address-line1').value = ''
+  addressSelector.querySelector('.address-line2').value = ''
+  addressSelector.querySelector('.address-town-or-city').value = ''
+  addressSelector.querySelector('.postcode').value = ''
+}
+
+function isAnyAddressFieldPopulated(postcodeLookupWidget) {
+  var addressSelector = postcodeLookupWidget.querySelector('.address')
+
+  return addressSelector.querySelector('.address-line1').value !== '' ||
+    addressSelector.querySelector('.address-line2').value !== '' ||
+    addressSelector.querySelector('.address-town-or-city').value !== '' ||
+    addressSelector.querySelector('.postcode').value !== ''
+}
+
+function showAddressError(isNorthernIrelandPostcode, postcodeLookupWidget) {
+  var northernIrelandErrorMessage = postcodeLookupWidget.querySelector('.postcode-search-error-ni')
+  var genericErrorMessage = postcodeLookupWidget.querySelector('.postcode-search-error')
+
+  if (isNorthernIrelandPostcode) {
+    northernIrelandErrorMessage.classList.remove('hidden')
+    genericErrorMessage.classList.add('hidden')
+  } else {
+    northernIrelandErrorMessage.classList.add('hidden')
+    genericErrorMessage.classList.remove('hidden')
+  }
+
+  postcodeLookupWidget.querySelector('.postcode-search-container').classList.add('form-group-error')
+}
+
+function lookupPostcode (postcode, postcodeLookupWidget) {
+  var xhr = new XMLHttpRequest()
+  xhr.open('GET', '/postcode-lookup?postcode=' + encodeURIComponent(postcode))
```

Comment:
maybe we can add some object to handle Ajax requests? probably there will be more and more scripts like this one?

Reply:
eg:
https://github.com/yarpo/yarpobject/blob/master/ajax/yAjax/yajax.js

Reply:
no, YAGNI

Reply:
Maybe this should be in some config?

```
/postcode-lookup?postcode=
```

Reply:
why? we control that in the frontend. if someone needs to change it they can.
I don't see any value in extracting that out and a lot of extra unneeded work to load the config from the backend and extra ajax calls etc